sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -452,22 +455,173 @@ def fetch_all_values(cls, cursor, counters_list, logger):
+            replica_server_name = row[replica_server_name_index]
+
+            metric_tags = [
+                'synchronization_state_desc:{}'.format(str(sync_state_desc)),
+                'replica_server_name:{}'.format(str(replica_server_name)),
+                'availability_group:{}'.format(str(resource_group_id)),
+            ]
+            metric_tags.extend(self.tags)
+            metric_name = '{}'.format(self.datadog_name)
+
+            self.report_function(metric_name, column_val, tags=metric_tags)
+
+
+# https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-hadr-availability-group-states-transact-sql?view=sql-server-ver15
+class SqlAvailabilityGroups(BaseSqlServerMetric):
+    TABLE = 'sys.dm_hadr_availability_group_states'
+    DEFAULT_METRIC_TYPE = 'gauge'
+    QUERY_BASE = 'select * \
+    from {table} as dhdrcs \
+    inner join sys.availability_groups as ag \
+    on ag.group_id = dhdrcs.group_id'.format(
+        table=TABLE
+    )
+
+    @classmethod
+    def fetch_all_values(cls, cursor, counters_list, logger):
+        return cls._fetch_generic_values(cursor, None, logger)
+
+    def fetch_metric(self, rows, columns):
+        value_column_index = columns.index(self.column)
+
+        resource_group_id_index = columns.index('resource_group_id')
+        sync_health_desc_index = columns.index('synchronization_health_desc')
+
+        for row in rows:
+            resource_group_id = row[resource_group_id_index]
+            selected_ag = self.cfg_instance.get('availability_group')
+
+            if selected_ag and selected_ag != resource_group_id:
+                continue
+
+            column_val = row[value_column_index]
+            sync_health_desc = row[sync_health_desc_index]
+            metric_tags = [
+                'availability_group:{}'.format(str(resource_group_id)),
+                'synchronization_health_desc:{}'.format(str(sync_health_desc)),
+            ]
+            metric_tags.extend(self.tags)
+            metric_name = '{}'.format(self.datadog_name)
+            selected_ag = self.cfg_instance.get('availability_group')
+
+            self.report_function(metric_name, column_val, tags=metric_tags)
+
+
+# https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-availability-replicas-transact-sql?view=sql-server-ver15
+class SqlAvailabilityReplicas(BaseSqlServerMetric):
+    TABLE = 'sys.availability_replicas'
+    DEFAULT_METRIC_TYPE = 'gauge'
+    QUERY_BASE = 'SELECT * FROM {table} as ar \
+                    inner join sys.dm_hadr_database_replica_cluster_states as dhdrcs \
+                    on ar.replica_id = dhdrcs.replica_id \
+                    inner join sys.dm_hadr_database_replica_states as dhdrs \
+                    on ar.replica_id = dhdrs.replica_id \
+                    inner join sys.availability_groups as ag \
+                    on ag.group_id = ar.group_id'.format(
+        table=TABLE
+    )
```

Comment:
Should try to make the `QUERY_BASE` text consistent across the classes in this PR (other classes use the convention of having everything lower-case, for better or worse).